raiden/transfer/utils.py

```diff
@@ -94,10 +95,14 @@ def encrypt_secret(
     return encrypted_secret
 
 
-def decrypt_secret(encrypted_secret: EncryptedSecret, private_key: PrivateKey) -> Secret:
+def decrypt_secret(
+    encrypted_secret: EncryptedSecret, private_key: PrivateKey
+) -> Tuple[Secret, PaymentAmount, PaymentID]:
     try:
         secret_dict = json.loads(decrypt(private_key, encrypted_secret).decode())
-        secret = Secret(decode_hex(secret_dict["secret"]))
     except (ValueError, json.JSONDecodeError):
         raise InvalidSecret
-    return secret
+    secret = Secret(decode_hex(secret_dict["secret"]))
+    amount = PaymentAmount(secret_dict["amount"])
```

Comment:
This should be fine to receive both str and int values, but we should ensure the object `to_encrypt` is serialised in a consistent format, i.e. numeric strings (for `amount` and `payment_identifier`)